### ISSUE TITLE
noUnusedLocals incorrectly flags Angular2 subscribe() method

### ISSUE THREAD

**Issue by atalis** [Microsoft/TypeScript#13961]
**TypeScript Version:**  2.0.0
We upgraded our TSLint version recently, and the new version deprecated the check for unused variables, because TypeScript compiler now has an option to check for that.  So we enabled the noUnusedLocals compiler option in tsconfig.json and now are seeing errors all over the place where they should not be.

**Code**
When you subscribe to an Angular 2 observable, you handle the success/error by mapping the corresponding callbacks to a function:

```ts
this.myService.performAction()
      .subscribe(
      data => {
         // do something with the returned data
      },
      error => {
         // process the error
      });

```
Sometimes all you want to do on success is log it or notify the object's listeners that the performed action has completed:

```ts
this.myService.performAction()
      .subscribe(
      data => {
        this.notify.emit();
      },
      error => {
         // process the error
      });

```

**Expected behavior:**
This is valid code, even though "data" is not used.  

**Actual behavior:**
I get an error:
   `'data' is declared but never used.`

I know that I can rename "data" to "_data" to ignore the warning, but I would prefer to keep our code consistent, regardless whether the "data" is or isn't used (it *is* used in most places).  At this point the only way for us to have the noUnusableLocals rule enforced *and* to have consistent code is to prefix all callbacks ("data", "error", etc.) with an underscore, even though stylistically it makes no sense (and would be inconsistent with the rest of the code to boot).

As this is valid format, the compiler should not be flagging the callback functions as unused.

**Comment by RyanCavanaugh** [comment]
You could write `() => { this.notify.emit(); }` instead. Or just turn off the option? The middle ground of "I want a flag that tells me about unused locals except some unused locals are OK" isn't really tenable.

**Comment by atalis** [comment]
Ah, ok, I didn't realize that I could use () instead of data.  Thanks, that'll work!
